sdk/identity/azure-identity-broker/azure/identity/broker/_browser.py

```diff
@@ -0,0 +1,155 @@
+        the Web Account Manager (WAM) as its authentication broker. If this parameter is set to True, the broker will
+        be used when possible. Defaults to False.
+        Check https://learn.microsoft.com/azure/active-directory/develop/scenario-desktop-acquire-token-wam for more
+        information about WAM.
+    :keyword int parent_window_handle: If your app is a GUI app running on a modern Windows system,
+        and your app opts in to use broker via `allow_broker`, you are required to also provide its window handle,
+        so that the sign in UI window will properly pop up on top of your window.
+    :keyword bool enable_msa_passthrough: Determines whether Microsoft Account (MSA) passthrough is enabled. Note, this
+        is only needed for select legacy first-party applications. Defaults to False.
+    :keyword bool disable_instance_discovery: Determines whether or not instance discovery is performed when attempting
+        to authenticate. Setting this to true will completely disable both instance discovery and authority validation.
+        This functionality is intended for use in scenarios where the metadata endpoint cannot be reached, such as in
+        private clouds or Azure Stack. The process of instance discovery entails retrieving authority metadata from
+        https://login.microsoft.com/ to validate the authority. By setting this to **True**, the validation of the
+        authority is disabled. As a result, it is crucial to ensure that the configured authority host is valid and
+        trustworthy.
+    :raises ValueError: invalid **redirect_uri**
+
+    .. admonition:: Example:
+
+        .. literalinclude:: ../samples/credential_creation_code_snippets.py
+            :start-after: [START create_interactive_browser_credential]
+            :end-before: [END create_interactive_browser_credential]
+            :language: python
+            :dedent: 4
+            :caption: Create an InteractiveBrowserCredential.
+    """
+
+    def __init__(self, **kwargs: Any) -> None:
+        self._allow_broker = kwargs.pop("allow_broker", None)
+        self._parent_window_handle = kwargs.pop("parent_window_handle", None)
+        self._enable_msa_passthrough = kwargs.pop("enable_msa_passthrough", False)
+        redirect_uri = kwargs.pop("redirect_uri", None)
+        if redirect_uri:
+            self._parsed_url = urlparse(redirect_uri)
+            if not (self._parsed_url.hostname and self._parsed_url.port):
+                raise ValueError('"redirect_uri" must be a URL with port number, for example "http://localhost:8400"')
+        else:
+            self._parsed_url = None
+
+        self._login_hint = kwargs.pop("login_hint", None)
+        self._timeout = kwargs.pop("timeout", 300)
+        client_id = kwargs.pop("client_id", DEVELOPER_SIGN_ON_CLIENT_ID)
+        super(InteractiveBrowserBrokerCredential, self).__init__(client_id=client_id, **kwargs)
```

Comment:
Parent constructor handles all this logic, so can remove most of this here.

```suggestion
        super().__init__(**kwargs)
```